scripts/generate-oas.ts

```diff
@@ -37,39 +37,69 @@ function kebabToCamelCase(str: string): string {
   );
 }
 
-export function resolveApiName(name: string): string {
-  // Special cases for shopper-orders and shopper-seo where the API name has different casing then the name in exchange.json
-  if (name === 'Shopper orders OAS') {
-    return 'ShopperOrders';
+function appendVersionIfV2(name: string, version: string): string {
+  if (version === 'V1' || version === 'v1') {
+    return name;
   }
-  if (name === 'Shopper Seo OAS') {
-    return 'ShopperSEO';
+  return name + version;
+}
+
+export function resolveApiName(name: string, version: string): string {
+  let apiName;
+
+  // Special handling for ShopperBasketV1 because we want to append V1 in this case
+  if (name === 'Shopper Baskets OAS' && version === 'v1') {
+    return 'ShopperBasketsV1';
   }
-  if (name === 'Shopper Context OAS') {
-    return 'ShopperContexts';
+
+  if (name === 'Shopper Seo OAS') {
+    apiName = 'ShopperSEO';
+  } else if (name === 'Shopper Context OAS') {
+    apiName = 'ShopperContexts';
+  } else {
+    apiName = name.replace(/\s+/g, '').replace('OAS', '');
   }
-  return name.replace(/\s+/g, '').replace('OAS', '');
+
+  return version !== 'v1' ? apiName + version.toUpperCase() : apiName;
```

Comment:
This seem to be opposite of what you are doing for Shopper Baskets? Why do we need this?

Reply:
This was my adjustment to allow for a more generic v2 as per your comment above.

Here, if `apiVersion` in exchange.json is v1, we don't append V1. This results in just the api name being returned. Ie. `ShopperProducts`

If the a`apiVersion` in exchange.json is v2, this returns `ShopperProductsV2`.

ShopperBaskets is a special case where the API team have set `x-sdk-classname` to ShopperBasketsV1 in the `-internal` OAS spec. So the API name returned here has to match that (ie. ShopperBasketsV1).

We don't want to append `V1` to other APIs (ie. we don't want ShopperProductsV1).